pkg/cloud/cloud.go

```diff
@@ -674,41 +659,12 @@ func (c *cloud) WaitForAttachmentState(ctx context.Context, volumeID, expectedSt
 			return false, nil
 		}
 
-		if len(volume.Attachments) > 1 {
-			// Shouldn't happen; log so we know if it is
-			klog.InfoS("Found multiple attachments for volume", "volumeID", volumeID, "volume", volume)
-		}
-		attachmentState := ""
+		attachmentState := volumeDetachedState
+
 		for _, a := range volume.Attachments {
-			if attachmentState != "" {
-				// Shouldn't happen; log so we know if it is
-				klog.InfoS("Found multiple attachments for volume", "volumeID", volumeID, "volume", volume)
-			}
-			if a.State != nil {
+			if a.State != nil && aws.StringValue(a.InstanceId) == expectedInstance && aws.StringValue(a.Device) == expectedDevice {
```

Comment:
Huge readability win by removing all that code, thank you!

Was going to comment about extracting this conditional out into it's own function (because I had to linger on it for a moment), but couldn't think of a good name so never-mind.